firestore/src/test/java/com/example/firestore/BaseIntegrationTest.java

```diff
@@ -32,9 +32,21 @@
  */
 public class BaseIntegrationTest {
 
-  protected static String projectId = "java-docs-samples-firestore";
+  protected static final String projectId = detectProjectId();
   protected static Firestore db;
 
+  private static String detectProjectId() {
+    String id = System.getProperty("firestore.project.id");
+    if (id != null) {
+      return id;
+    }
+    id = System.getenv("FIRESTORE_PROJECT_ID");
+    if (id != null) {
+      return id;
+    }
+    return "java-docs-samples-firestore";
+  }
+
```

Comment:
```
    public static String detectProjectId() {
        String projectId = System.getenv("FIRESTORE_PROJECT_ID");
        projectId = projectId != null ? projectId : System.getProperty("firestore.project.id", "java-docs-sample-firestore");
        return projectId;
    }
```
what about this?

Reply:
value set via system property should override value set via env variable. In your case env variable will override system prop.